### PR TITLE
task: customize touchable component

### PR DESCRIPTION
**Pull request by ManuViola77**
### Pull request type

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

---

### Description

This PR customizes the Touchable component according to the designs. It has 3 variants, Primary, Secondary and Tertiary.

---

#### Jira board reference

- [Create Basic Components](https://rootstrap.atlassian.net/browse/TH-111)

---

#### Notes:

- TS gives an error when I try to change the Icon's Tint Color, so for now that part is commented.

---

#### Tasks:

- [x] Tested on IOS
- [x] Tested on Android
- [ ] Tested on a small device
- [ ] Tested on a real device
- [ ] Tested all flows related with this PR changes
- [ ] Tested accessibility
- [ ] Added tests

---

#### Preview

iOS:

https://user-images.githubusercontent.com/8755889/224463908-42b54a53-6a38-4bd7-82bb-617de7abc8d7.mp4

Android:

### CODE REVIEW

File: src/design-system/components/Touchable/styles.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import { Styles } from 'dripsy';
+
+export default Styles.create({
+  container: {
+    alignItems: 'center',
+    borderRadius: 16,
+    marginHorizontal: 16,
```
Comment (jpirazusta) [inline src/design-system/components/Touchable/styles.tsx:7]:
```suggestion
    marginX: 16,
```

File: src/design-system/components/Touchable/styles.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    backgroundColor: 'primary-500',
+  },
+  primaryDisabled: {
+    backgroundColor: 'n-100',
+  },
+  primaryDisabledText: {
+    color: 'n-400',
+  },
+  primaryIcon: {
+    // tintColor: 'white', // TODO: check how to assign tint color without TS error
```
Comment (jpirazusta) [inline src/design-system/components/Touchable/styles.tsx:20]: For the `tintColor` in Ensmbl we used the `StyleSheet` from `react-native`. Something like:

`import { theme } from 'design-system/theme';
import { Styles } from 'dripsy';
import { StyleSheet } from 'react-native';

const stylesRN = StyleSheet.create({
  tintColor: {
    tintColor: theme.colors.background,
  },
});

const styles = Styles.create({
  // Something
});

export default { ...styles, ...stylesRN };
`

**Review by jpirazusta — APPROVED** [review]
🙌

File: src/design-system/components/Touchable/styles.tsx (hunk 2 above)
Comment (nicmalegre) [reply src/design-system/components/Touchable/styles.tsx:20]: I think we have 2 solutions here:

1. The one mentioned by JP, using StyleSheet from 'react-native'
2. Creating a function that given a `color name` return that color from the current theme combined with some TS . Here is the example: https://github.com/UNPRESS-NEWS/unpress-mobile/blob/0b08c8c94320d42403e8043fa34f239226149b10/src/components/common/rich-text/RichTextEditorToolbar/styles.ts#L16

Comment (ManuViola77) [reply src/design-system/components/Touchable/styles.tsx:20]: oh great, awesome!  I will do this tint color in another PR, ok?

**Review by nicmalegre — APPROVED** [review]

**Pull request merged**